sw/device/silicon_creator/manuf/lib/sram_exec_test.c

```diff
@@ -0,0 +1,82 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/pinmux_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/status.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_uart_t uart0;
+static dif_pinmux_t pinmux;
+static dif_otp_ctrl_t otp;
+
+static const uint8_t kNumDeviceId = 8;
+
+/**
+ * Read and return 32-bit OTP data via the DAI interface.
+ */
+static status_t otp_ctrl_dai_read_32(const dif_otp_ctrl_t *otp,
```

Comment:
can you not just use the `otp_ctrl_testutils_dai_read32()` in the `otp_ctrl_testutils` lib? Did you want to avoid bringing in the whole testutils lib for code size?

Reply:
I was doing some tests and trying to make the SRAM binary small because loading a big binary was causing problems. This will be fixed in the final version.

Reply:
ahh makes sense, yeah that doesn't make sense to be static, you can change it.